syntax = "proto3";

message ShaderConfig {
  string name = 1;
  string vs_path = 2;
  string fs_path = 3;
  optional string gs_path = 4;
  optional string ts_path = 5;
}

enum TextureType {
  Texture2D = 0;
  CubeMap = 1;
}

message TextureConfig {
  string name = 1;
  repeated string paths = 2;
  TextureType texture_type = 3;
}

message Config {
  repeated ShaderConfig shader_config = 1;
  repeated TextureConfig texture_config = 2;
}